crates/router/src/core/utils.rs

```diff
@@ -543,6 +574,43 @@ pub async fn construct_retrieve_file_router_data<'a>(
         reference_id: None,
         payment_method_token: None,
         preprocessing_id: None,
+        connector_request_reference_id: "irrelevant_connector_request_reference_id_in_dispute_flow"
+            .to_string(),
     };
     Ok(router_data)
 }
+
+pub async fn get_connector_request_reference_id(
+    db: &dyn db::StorageInterface,
+    merchant_id: &str,
+    payment_attempt: &storage_models::payment_attempt::PaymentAttempt,
+) -> String {
+    let lookup_key = "send_payment_id_as_connector_request_reference_id";
+    let option_config_map = match db.find_config_by_key(lookup_key).await {
```

Comment:
Should this be stored in redis instead, as this db call will happen in all the flows where connector is called.

Reply:
Also, how can the merchant update this field? or is it us who have to update it as requested by the merchant.

Reply:
We have to update this field while onboarding a merchant